app/uk/gov/hmrc/teamsandrepositories/connectors/BranchProtectionApiConnector.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.teamsandrepositories.connectors
+
+import com.amazonaws.auth.AWSCredentialsProvider
+import io.ticofab.AwsSigner
+import play.api.Logging
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+import uk.gov.hmrc.http.{HeaderCarrier, _}
+import uk.gov.hmrc.teamsandrepositories.config.BranchProtectionApiConfig
+
+import java.time.LocalDateTime
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+import uk.gov.hmrc.http.HttpReads.Implicits._
+
+@Singleton
+class BranchProtectionApiConnector @Inject()(
+  httpClient: HttpClient,
+  awsCredentialsProvider: AWSCredentialsProvider,
+  config: BranchProtectionApiConfig
+)(
+  implicit ec: ExecutionContext
+) extends Logging {
+
+  private implicit val hc = HeaderCarrier()
+
+  def setBranchProtection(repoName: String): Future[Unit] = {
+    val url =
+      url"${config.baseUrl}/v1/UpdateGithubDefaultBranchProtection"
+
+    val payload =
+      Json.toJson(BranchProtectionApiConnector.Request(repoName, enable = true))
+
+    val signedHeaders =
```

Comment:
is it worth adding an AWSSigner as we did [here](https://github.com/hmrc/self-service-secrets-frontend/blob/main/app/uk/gov/hmrc/selfservicesecretsfrontend/connectors/AWSSigner.scala#L18)?
It can be [used](https://github.com/hmrc/self-service-secrets-frontend/blob/main/app/uk/gov/hmrc/selfservicesecretsfrontend/connectors/TenantSecretServiceAPIConnector.scala#L79) to show clearly that the method,url,headers,payloads all align correctly